pkg/virt-controller/network/ipamclaims.go

```diff
@@ -0,0 +1,194 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+
+	"kubevirt.io/client-go/kubecli"
+
+	ipamclaims "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+)
+
+type IPAMClaimParams struct {
+	claimName   string
+	networkName string
+}
+
+func GetNetworkToIPAMClaimParams(virtClient kubecli.KubevirtClient, namespace string, vmiName string, networks []virtv1.Network) (map[string]IPAMClaimParams, error) {
+	networkToIPAMClaimName := make(map[string]IPAMClaimParams)
+	for _, network := range vmispec.FilterMultusNonDefaultNetworks(networks) {
+		if persistentIPsNetworkName, err := getPersistentIPsNetworkName(virtClient, namespace, network.Multus.NetworkName); err != nil {
+			return map[string]IPAMClaimParams{}, fmt.Errorf("failed retrieving persistentIPsNetworkName: %w", err)
+		} else if persistentIPsNetworkName != "" {
+			networkToIPAMClaimName[network.Name] = IPAMClaimParams{
+				claimName:   fmt.Sprintf("%s.%s", vmiName, network.Name),
+				networkName: persistentIPsNetworkName,
+			}
+		}
+	}
+
+	return networkToIPAMClaimName, nil
+}
```

Comment:
This should be `GetVirtualMachineNetworkToCNINetworkName` and we just need it to return `map[string]string`